packages/sdk/src/client/index.ts

```
@@ -0,0 +1,668 @@
+			};
+		}
+
+		return this.errorResult(response);
+	};
+
+	private httpResponseToMutationResult = async <Data>(response: Response): Promise<MutationResult<Data>> => {
+		if (this.isOK(response)) {
+			const responseJSON = await response.json();
+			if (responseJSON.errors && !responseJSON.data) {
+				return {
+					status: 'error',
+					errors: responseJSON.errors,
+				};
+			}
+			if (responseJSON.errors && responseJSON.data) {
+				return {
+					status: 'partial',
+					errors: responseJSON.errors,
+					data: responseJSON.data,
+				};
+			}
+			return {
+				status: 'ok',
+				data: responseJSON.data,
+			};
+		}
+
+		return this.errorResult(response);
+	};
+
+	private jsonToSubscriptionResult = <Data>(json: any): SubscriptionResult<Data> => {
+		const responseJSON = JSON.parse(json);
+		if (responseJSON.errors && !responseJSON.data) {
+			return {
+				status: 'error',
+				errors: responseJSON.errors,
+			};
+		}
+		if (responseJSON.errors && responseJSON.data) {
+			return {
+				status: 'partial',
+				errors: responseJSON.errors,
+				data: responseJSON.data,
+			};
+		}
+		if (responseJSON.data) {
+			return {
+				status: 'ok',
+				streamState: 'streaming',
+				data: responseJSON.data,
+			};
+		}
+		return {
+			status: 'error',
+			errors: [{ message: 'Unexpected response' }],
+		};
+	};
+
+	public fetchUser = async (abortSignal?: AbortSignal, revalidate?: boolean): Promise<User<Role> | null> => {
+		try {
+			const revalidateTrailer = revalidate === undefined ? '' : '?revalidate=true';
+			const response = await fetch(
+				this.baseURL + '/' + this.applicationPath + '/auth/cookie/user' + revalidateTrailer,
+				{
+					headers: {
+						...this.extraHeaders,
+						'Content-Type': 'application/json',
+						'WG-SDK-Version': this.sdkVersion,
+					},
+					method: 'GET',
+					credentials: 'include',
+					mode: 'cors',
+					signal: abortSignal,
+				}
+			);
+			if (this.isOK(response)) {
+				return response.json();
+			}
```

As a consumer, I'd expect to get an error rather than nothing. Let's return the response or some information for debugging?

---

I suppose we should throw an error in this case. But this brings the next question, should we throw every where instead of returning a result with the error message.

---

Generally, yes. But for now, I'd try to do it only for low-level methods like the client and handle it in the same way as now in the consumer packages (next.js, react ...). So if someone uses this client package, he has full control over handling an error.

---

Introducing this will break a thing or two in the next.js client. I wanted to prevent that in this PR. And only introduce  any breaking changes in the new React package.

---

Could you point me to the relevant places?

---

https://github.com/wundergraph/wundergraph/blob/main/packages/nextjs/src/index.ts#L797

This is also mentioned in a blog post somewhere, so that should be updated too if we change this.

But.. it already catches any error errors in this code, perhaps the change is as easy as `setUser(null)` in the catch, in case the call fails.

---

Blog post is irrelevant. We can't update it after every change :smile: 

> But.. it already catches any error errors in this code, perhaps the change is as easy as setUser(null) in the catch, in case the call fails.

Yes, that should do the trick.